src/writer/serve.py

```
@@ -272,13 +280,38 @@ async def import_zip(file: UploadFile = File(...)):
+            logging.info("[Import Debug] Calling app_runner.import_zip")
+            
             await app_runner.import_zip(tmp_path)
-            os.remove(tmp_path)
-        except ValueError:
-            raise HTTPException(status_code=400, detail="Invalid upload.")
+            
+            logging.info("[Import Debug] app_runner.import_zip completed successfully")
+            return {"status": "success", "message": "Import completed successfully"}
+        except ValueError as e:
+            logging.error("[Import Debug] ValueError: %s", e, exc_info=True)
+            raise HTTPException(status_code=400, detail=f"Invalid upload: {str(e)}")
```

Ruff (0.14.5)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)